benchmarks/src/main/java/org/elasticsearch/benchmark/xcontent/FilterContentBenchmark.java

```
@@ -0,0 +1,182 @@
+        }
+
+        Set<String> includes;
+        Set<String> excludes;
+        if (inclusive) {
+            includes = filters;
+            excludes = null;
+        } else {
+            includes = null;
+            excludes = filters;
+        }
+        parserConfig = XContentParserConfiguration.EMPTY.withFiltering(includes, excludes);
+    }
+
+    @Benchmark
+    public BytesReference filterWithParserConfigCreated() throws IOException {
+        try (BytesStreamOutput os = new BytesStreamOutput()) {
+            XContentBuilder builder = new XContentBuilder(XContentType.JSON.xContent(), os);
+            try (XContentParser parser = XContentType.JSON.xContent().createParser(parserConfig, source.streamInput())) {
+                builder.copyCurrentStructure(parser);
+                return BytesReference.bytes(builder);
+            } catch (Exception e) {
+                return new BytesArray("");
+            }
+        }
+    }
+
+    @Benchmark
+    public BytesReference filterWithoutParserConfigCreated() throws IOException {
+        try (BytesStreamOutput os = new BytesStreamOutput()) {
+            XContentBuilder builder = new XContentBuilder(XContentType.JSON.xContent(), os);
+            Set<String> includes;
+            Set<String> excludes;
+            if (inclusive) {
+                includes = filters;
+                excludes = null;
+            } else {
+                includes = null;
+                excludes = filters;
+            }
+            XContentParserConfiguration contentParserConfiguration = XContentParserConfiguration.EMPTY.withFiltering(includes, excludes);
+            try (XContentParser parser = XContentType.JSON.xContent().createParser(contentParserConfiguration, source.streamInput())) {
+                builder.copyCurrentStructure(parser);
+                return BytesReference.bytes(builder);
+            } catch (Exception e) {
+                return new BytesArray("");
```

Could you let this bubble out? That'd make it abort the benchmark and I think we want that.

---

because of the bug in https://github.com/elastic/elasticsearch/issues/80142. some tests will failed